marlowe-dashboard-client/src/Pickup/View.purs

```diff
@@ -49,140 +49,157 @@ renderPickupCard pickupCard newWalletNicknameKey wallets =
   in
-    div
-      [ classNames [ "flex", "flex-col" ] ]
-      [ h2_
-          [ text "New wallet generated" ]
-      , label
-          [ for "newWalletKey" ]
-          [ text "Key:" ]
-      , input
-          [ type_ InputText
-          , id_ "newWalletKey"
-          , value key
-          , readOnly true
-          ]
-      , br_
-      , label
-          [ for "newWalletNickname" ]
-          [ text "Nickname:" ]
-      , input
-          [ type_ InputText
-          , classNames $ toggleWhen (mNicknameError == Nothing) "border-green" "border-red"
-          , id_ "newWalletNickname"
-          , placeholder "Nickname"
-          , value nickname
-          , onValueInput_ SetNewWalletNickname
+    div_
+      [ p
+          [ classNames [ "font-bold", "mb-1" ] ]
+          [ text if new then "Play wallet generated" else "Play wallet " <> nickname ]
+      , div
+          [ classNames $ Css.hasNestedLabel <> [ "mb-1" ] ]
+          $ [ label
+                [ classNames $ Css.nestedLabel
+                , for "newWalletNickname"
+                ]
+                [ text "Nickname" ]
+            , input
+                $ [ type_ InputText
+                  , classNames $ Css.input $ if new then isJust mNicknameError else false
+                  , id_ "newWalletNickname"
+                  , placeholder "Nickname"
+                  , value nickname
+                  ]
+                <> if new then [ onValueInput_ SetNewWalletNickname ] else [ readOnly true ]
+            ]
+          <> if new then
+              [ div
+                  [ classNames Css.inputError ]
+                  $ case mNicknameError of
+                      Just nicknameError -> [ text $ show nicknameError ]
+                      Nothing -> []
+              ]
+            else
+              []
+      , div
+          [ classNames $ Css.hasNestedLabel <> [ "mb-1" ] ]
+          [ label
+              [ classNames Css.nestedLabel
+              , for "newWalletKey"
+              ]
+              [ text "Public key" ]
+          , input
+              [ type_ InputText
+              , classNames $ Css.input false
+              , id_ "newWalletKey"
+              , value key
+              , readOnly true
+              ]
           ]
       , div
-          [ classNames [ "mb-1", "text-red", "text-sm" ] ]
-          $ case mNicknameError of
-              Just nicknameError -> [ text $ show nicknameError ]
-              Nothing -> []
-      , button
-          [ disabled $ isJust mNicknameError
-          , onClick_ PickupNewWallet
+          [ classNames [ "flex" ] ]
+          [ button
+              [ classNames $ Css.secondaryButton <> [ "flex-1", "mr-1" ]
+              , onClick_ $ SetCard Nothing
+              ]
+              [ text "Cancel" ]
+          , button
+              [ classNames $ Css.primaryButton <> [ "flex-1" ]
+              , disabled $ if new then isJust mNicknameError else false
```

Comment:
`disabled (new && isJust mNicknameError)`

Reply:
Yeah, I take your point. Thing is this card is essentially two slightly different cards - one for when it's a new wallet, and one for when it's an existing wallet. So I've just got this `if new then ... else ...` pattern in various places, which I feel is clearer given the context. 😕 

Maybe this is the wrong approach overall though. Maybe I should have a specific data type for these two cases instead of using a Boolean. 🤔